src/nvim/testdir/test_stat.vim

```diff
@@ -39,11 +39,14 @@ func Test_existent_file()
 endfunc
 
 func Test_existent_directory()
+  if has('win32')
+    throw 'skipped: Nvim does not treat directories as executable on Windows'
+  endif
   let dname = '.'
 
   call assert_equal(0, getfsize(dname))
   call assert_equal('dir', getftype(dname))
-  call assert_equal(has('win32') ? 'rw-' : 'rwx', getfperm(dname)[0:2])
```

Comment:
Let's keep this. I just looked more closely, and found that Vim's `getfperm` is inconsistent with `executable()`.